### ISSUE TITLE
Vulkan unexpected map rendering behavior 

### ISSUE THREAD

**Issue by sctt** [ddnet/ddnet#6319]
Users are reporting issues on our map since Vulkan been made default on DDNet clients.
![renderer-bug](https://user-images.githubusercontent.com/3328841/215740797-3bbfe2ac-93ba-481e-9016-ed77be81baa4.png)

The problem arises only on enveloped quads with RGBA EnvPoints values greater than 1, and only in game, not in the editor.
Clapping all env points to 1 doesn't completely fix the issue, as Quad Color * Env Color with EnvColor > 1 doesn't necessarly produce a capped 255 value, thus every single bugged quad should be fixed individually.

Even thought RGBA > 1 on Envs might be seen as a mapping error by someone who might say this is a wrong env configuration and should be avoided,  i personally don't completely agree with that, i still see this as a client bug since:
1. i expect the editor not to allow me to produce wrong and buggy configurations that easily, like it does for everything else.
2. i expect the editor to represent the same picture im going to see in game, not this is lack of reliability.
3. if the previous default renderer didn't have this problem, it was better at least in handling this situation. is there an explanation on why the new one isn't doing the same?

i'd like to hear from you if this can be fixed by its root cause, or if i have to write a map_fix_envs tool to clean the map (i'd prefer to avoid wasting time if there is a better solution, like correctly rendering those values like the previous renderer did).

thank you.

**Comment by Jupeyy** [comment]
Editor still uses the old rendering technique, but please adopt to the new behavior, it makes more sense

You can use patigas python binding to fix it for all envelops

**Comment by Jupeyy** [comment]
I'll look into removing the clamp for old rendering.

U can ask poppy on discord to give u the fix too. I think he worked on a map that looks the same

**Comment by sctt** [comment]
> I'll look into removing the clamp for old rendering.
> 
> U can ask poppy on discord to give u the fix too. I think he worked on a map that looks the same

thanks, i'll ask for the fix then, i'm glad someone already made it.
for the rest, it's very easy to produce this problem by mistake, mappers shouldn't even worry about this i think.
i think we should clamp the env values on edit, or at least print a warning....

**Comment by Jupeyy** [comment]
I'll later see if clamping is really good. It could be useless to not clamp. But i fully understand why u day that. Maybe the editor should have an option to disable the clamp or smth

From a pure math point a rgb(0.5, 0,5 0,5) quad multiplied by 2 should be fully white.

**Comment by sctt** [comment]
> I'll later see if clamping is really good. It could be useful to not clamp. But i fully understand why u say that. Maybe the editor should have an option to disable the clamp or smth
> 
> From a pure math point a rgb(0.5, 0,5 0,5) quad multiplied by 2 should be fully white.

i agree on what you are saying too, but mappers don't care of math. they play with values, they see the result, they expect the same on game: that's why is easy to create this problem, because of inconsistency.
if it's easier to change the editor to make it render the new way, it's true we wouldn't need any clamp or warning or any other workaround

**Comment by AssassinTee** [comment]
The current ddnet editor doesn't even allow alpha values > 1 (or < 0). Please update the map if you're running into this problem, the old behavior didn't make much sense and I don't ever see us switching back to the old behavior
